core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -1206,4 +1206,61 @@ public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
       return ExprEval.of(expr.value() != null, ExprType.LONG);
     }
   }
+
+  class LpadFunc implements Function
+  {
+    @Override
+    public String name()
+    {
+      return "lpad";
+    }
+
+    @Override
+    public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
+    {
+      if (args.size() != 3) {
+        throw new IAE("Function[%s] needs 3 arguments", name());
+      }
+      
+      String base = args.get(0).eval(bindings).asString();
+      int len = args.get(1).eval(bindings).asInt();
+      String pad = args.get(2).eval(bindings).asString();
+
+      if (base == null || pad == null) {
```

Comment:
I feel like `pad` being `null` should be an error case instead of silently returning `null`, but what do Oracle and Hive do in this case?

Reply:
Hmm, interesting, did a bit of searching and from the docs I've found, it seems like they have different behavior, [with Oracle having a default `pad` value of a single blank character](https://docs.oracle.com/cd/B28359_01/olap.111/b28126/dml_functions_2017.htm#OLADM593), and [Hive returning null](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+UDF) as you are doing here. If this is true, I'm not actually sure what is best thing to do here...

Reply:
Hi, @clintropolis , Postgres/Orcale has two func signatures which are two params or three params. When only having two params such as `lpad('abc',5)` it will pad with default blank char.

For three params, I tested postgres, oracle and hive when padding or base is `null`.
Postgres and hive:
```
select lpad('b',5,null) is null;  ##result is true
select lpad(null,5,'x') is null;  ##result is true
```
Oracle
```
create table test1(a int);
insert into test1(a) values(5);
select count(1) from test1 where lpad('abc',5,null) is null;# result is 1
select count(1) from test1 where lpad(null,5,'x') is null; #result is 1
select count(1) from test1 where lpad('abc',5) is null;   # result is 0
```
It looks the results are all `null` when base or pad is null and the same

Reply:
Thanks for testing! `null` seems like the thing to do then :+1: